File: src/DClare.Sdk.TypeScript/models/remote-agent-definition.ts
import { AgentCommunicationChannelDefinition } from './agent-communication-channel-definition';

/**
 * Represents the definition of a remote agent
 */
export class RemoteAgentDefinition
 {
  /**
   * Gets/sets the communication channel used to interact with the remote agent
   */
  channel: AgentCommunicationChannelDefinition;
}

File: src/DClare.Sdk/Models/RemoteAgentDefinition.cs
namespace DClare.Sdk.Models;

/// <summary>
/// Represents the definition of a remote AI agent.
/// </summary>
[Description("Represents the definition of a remote AI agent.")]
[DataContract]
public record RemoteAgentDefinition
{

    /// <summary>
    /// Gets or sets the communication channel used to reach the remote agent.
    /// </summary>
    [Description("The communication channel used to reach the remote agent.")]
    [Required]
    [DataMember(Name = "channel", Order = 1), JsonPropertyName("channel"), JsonPropertyOrder(1), YamlMember(Alias = "channel", Order = 1)]
    public virtual AgentCommunicationChannelDefinition Channel { get; set; } = null!;

}

